src/Fields/Fields.jl

```diff
@@ -45,11 +45,11 @@ comm_context(space::Spaces.ExtrudedFiniteDifferenceSpace) =
 comm_context(space::Spaces.SpectralElementSpace2D) =
     comm_context(space.topology)
 comm_context(space::S) where {S <: Spaces.AbstractSpace} =
-    ClimaComms.SingletonCommsContext()
+    comm_context(space.topology)
```

Comment:
could deprecate `comm_context` altogether, and overload `ClimaComms.context` (same we do for `ClimaComms.device`)?

Reply:
Isn't `comm_context` still a useful getter for users if all spaces contain a context?

Reply:
Oh, I misread, so change this function name to `context` @simonbyrne?

Reply:
Yeah, that would make sense, we can then just overload it.

Reply:
I mean overload `ClimaComms.context`